datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -0,0 +1,79 @@
+package com.ds.datastore;
+
+
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+@RestController
+public class BookController {
+
+    private final BookRepository repository;
+
+    private final BookModelAssembler assembler;
+
+    BookController(BookRepository repository, BookModelAssembler assembler) {
+
+        this.repository = repository;
+        this.assembler = assembler;
+    }
+
+    @PostMapping("/books")
+    Book newBook(@RequestBody Book book) {
+        return repository.save(book);
+    }
+    @GetMapping("/books/{id}")
+    EntityModel<Book> one(@PathVariable Long id) {
+        Book book = repository.findById(id).orElseThrow(() -> new BookNotFoundException(id));
```

Comment:
Currently, if the book is not found, it will return a `500` status code (Internal Server Error). 

If the book is not found, the convention is to return a `404` along with a helpful error message - so let's do that here.